pyproject.toml

```diff
@@ -53,6 +53,8 @@ optional = true
 [tool.poetry.group.test.dependencies]
 hypothesis = "^6.68.2"
 pytest = "^7.1.0"
+pytest-mock = "^3.10.0"
```

Comment:
I think we can avoid this added dependency with the `.cached_venv` class member I suggest above (and set in `test_resolver` below).